### PR TITLE
Feature/fix undefined xsd attributes and elements

### PR DESCRIPTION
**Pull request by edward-ellis**
This is the pull request we told you was coming today.  It was produced by @markborkum who is a new member of our team.

### CODE REVIEW

**Comment by jasondegraw** [comment]
@edward-ellis @markborkum We will get back to you soon with more of a review, but for now please roll back the changes in the example files directory. That is something that needs to be looked at, but it should be done separately.

**Comment by markborkum** [comment]
@jasondegraw When you say "roll back the changes in the example files directory," are you referring to: the added ".xml" suffixes and/or the modified `schemaLocation` attributes?

**Comment by jasondegraw** [comment]
@markborkum Yes, I think those are all of the changes inside the `examples` directory.

File: Gemfile [hunk 1]
```diff
@@ -5,4 +5,4 @@ gem 'nokogiri', '= 1.6.8.1'
 gem 'rubyXL'
 
 gem 'minitest'
-gem 'minitest-ci', git: 'git@github.com:circleci/minitest-ci.git'
+gem 'minitest-ci', git: 'https://github.com/circleci/minitest-ci.git', branch: 'master'
```
Comment (jasondegraw) [inline Gemfile]: @nllong This is in your area of expertise. Is this change OK with you?

Comment (nllong) [reply Gemfile]: I am exclusively a git@ protocol user, so I'm not sure the impact of this. Based on the number of warnings I have been getting around using `https`, I'm sure this is okay.

**Comment by edward-ellis** [comment]
We are going to withdraw this pull request and replace it with seven more finely grained requests that each address a specific issue.  That should make reviewing the issues simpler.

**Comment by jasondegraw** [comment]
@edward-ellis Well, that's your prerogative, but if the end result is something similar to what's done here, I encourage you to see this PR through. Since we've already reviewed this one, seven more will likely lead to delays.

File: src/data_dictionary.rb [hunk 2]
```diff
@@ -157,9 +157,10 @@ def generate
           doc_string = node.parent.parent.parent.parent.parent.xpath('.//xs:annotation/xs:documentation').first
         end
 
-        sub_name = node.parent.parent.parent.parent.parent.attribute('name').value
-        data[:sub_name] = sub_name
-        name += " (#{sub_name})"
+        if sub_name = node.parent.parent.parent.parent.parent.attribute('name')
+          data[:sub_name] = sub_name.value
+          name += " (#{sub_name.value})"
+        end
```
Comment (jasondegraw) [inline src/data_dictionary.rb]: @nllong Do these changes look OK to you?

**Pull request merged**